### PR TITLE
fix: Use Redis ConnectionManager to reconnect on failures

### PR DESCRIPTION
**Pull request by sbernauer**
Previously trino-lb would not recover from a closed connection

```
2024-06-21T08:00:17.924954Z  INFO Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Updated query counters from 1 remote clusters
2024-06-21T08:00:22.720512Z ERROR Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Failed to get last QueryCountFetcher update timestamp err=RedisError { source: GetLastQueryCountFetcherUpdate { source: broken pipe } }
2024-06-21T08:00:27.719737Z ERROR Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Failed to get last QueryCountFetcher update timestamp err=RedisError { source: GetLastQueryCountFetcherUpdate { source: broken pipe } }
2024-06-21T08:00:32.720312Z ERROR Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Failed to get last QueryCountFetcher update timestamp err=RedisError { source: GetLastQueryCountFetcherUpdate { source: broken pipe } }
```

Now the first call to redis will fail, but the following calls will work again

```
2024-06-21T07:58:57.264989Z  INFO Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Updated query counters from 1 remote clusters
2024-06-21T07:59:02.085295Z ERROR Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Failed to get last QueryCountFetcher update timestamp err=RedisError { source: GetLastQueryCountFetcherUpdate { source: broken pipe } }
2024-06-21T07:59:07.086128Z  INFO Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Updated query counters from 0 remote clusters
2024-06-21T07:59:12.085915Z  INFO Fetching current query counters: trino_lb::maintenance::query_count_fetcher: QueryCountFetcher: Updated query counters from 0 remote clusters
```

### CODE REVIEW

**Comment by Techassi** [comment]
How many times does the connection manager try to reconnect? Is there an exponential timeout? Are these parameters configurable?

Because these reconnect attempts can cause performance issues on the Redis server side which could lead to DoS.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,18 +4,23 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
-### Changed
+### Fixed
 
-- Bump Rust dependency versions ([#19]).
+- Use redis [`ConnectionManager`](https://docs.rs/redis/latest/redis/aio/struct.ConnectionManager.html) to re-connect on
```
Comment (maltesander) [inline CHANGELOG.md:9]:
```suggestion
- Use redis [`ConnectionManager`](https://docs.rs/redis/latest/redis/aio/struct.ConnectionManager.html) to reconnect on
```

**Comment by sbernauer** [comment]
Seems like they are using https://docs.rs/tokio-retry/latest/tokio_retry/ for that. They have some default values in https://docs.rs/redis/latest/src/redis/aio/connection_manager.rs.html#101-109, which I guess should be a good starting point

**Review by maltesander — APPROVED** [review]
LGTM if sascha is happy :)

**Comment by Techassi** [comment]
> Seems like they are using https://docs.rs/tokio-retry/latest/tokio_retry/ for that. They have some default values in https://docs.rs/redis/latest/src/redis/aio/connection_manager.rs.html#101-109, which I guess should be a good starting point

Thanks! That looks reasonable.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,18 +4,23 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
-### Changed
+### Fixed
```
Comment (Techassi) [inline CHANGELOG.md:7]: I assume removing the entry for the Rust dependency bumps was intentional?

**Review by maltesander — APPROVED** [review]
LGTM!

**Review by Techassi — APPROVED** [review]
LGTM, just one question (non-blocking).

File: CHANGELOG.md (hunk 2 above)
Comment (sbernauer) [reply CHANGELOG.md:7]: Jep, as it should not be user-facing

**Pull request merged**